libyul/backends/wasm/WasmDialect.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * Dialects for Wasm.
+ */
+
+#include <libyul/backends/wasm/WasmDialect.h>
+
+using namespace std;
+using namespace yul;
+
+WasmDialect::WasmDialect(AsmFlavour _flavour):
+	Dialect{_flavour}
+{
+	for (auto const& name: {
```

Comment:
Would it be better if we explicitly specify argument types? i.e instead of accepting `add`, we accept `addu32`, `adds32`, `addu64`, or `adds64`. (cf. #6548)

Reply:
This first prototype assumes `u64` everywhere. We plan to replace those by `u64.add`, `i64.add` just like in wasm itself.

Reply:
I understand that this is a very first prototype, but is there a downside to call it `u64.add` instead of `add`? (asked this just in case I'm missing something)

Reply:
Yes, the scanner currently does not allow `.` as part of an identifier, and we cannot easily change it because the scanner is shared between Solidity and Yul (for inline assembly).

Reply:
Ok it makes sense